irohad/network/impl/peer_communication_service_impl.cpp

```diff
@@ -27,18 +28,30 @@ namespace iroha {
     }
 
     void PeerCommunicationServiceImpl::propagate_transaction(
-        std::shared_ptr<const model::Transaction> transaction) {
+        std::shared_ptr<const shared_model::interface::Transaction>
+            transaction) {
       log_->info("propagate tx");
-      ordering_gate_->propagate_transaction(transaction);
+      ordering_gate_->propagate_transaction(
+          std::shared_ptr<model::Transaction>(transaction->makeOldModel()));
     }
 
-    rxcpp::observable<model::Proposal>
+    rxcpp::observable<std::shared_ptr<shared_model::interface::Proposal>>
     PeerCommunicationServiceImpl::on_proposal() {
-      return ordering_gate_->on_proposal();
+      return ordering_gate_->on_proposal().map(
+          [](auto prop) -> std::shared_ptr<shared_model::interface::Proposal> {
```

Comment:
explicit return type definition used to cast returned value with proto type to interface type